File: src/types/social.ts
export type EsgType = 'Environmental' | 'Social' | 'Governance'
export type SdgNumber = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12 | 13 | 14 | 15 | 16 | 17

export interface SocialActivity {
  id: number
  name: string
  organization: string
  esgType: EsgType
  sdgNumbers: SdgNumber[]
  periodFrom: string
  periodTo: string
  contribution: string
  reflection: string
  photoUrl?: string
}

File: backend/app/models/social.py
from sqlalchemy import Column, Integer, String, Text, Date
from ..database import Base


class SocialActivity(Base):
    __tablename__ = "social_activities"

    id           = Column(Integer, primary_key=True, index=True)
    name         = Column(String(200), nullable=False)
    organization = Column(String(200), nullable=False)
    esg_type     = Column(String(20), nullable=True)      # Environmental | Social | Governance；與 sdg_numbers 互斥（擇一分類）
    sdg_numbers  = Column(Text, default="[]")            # JSON array e.g. [1, 3, 13]
    period_from  = Column(Date, nullable=False)
    period_to    = Column(Date, nullable=True)
    reflection   = Column(Text, nullable=False, default="")
    youtube_url  = Column(String(500), nullable=True)
    photos       = Column(Text, default="[]")            # JSON array: [{url, position}]，最多 MAX_SOCIAL_PHOTOS 張
